examples/drag_and_drop/src/lib.rs

```diff
@@ -0,0 +1,198 @@
+    } else {
+        panic!("wrong id");
+    }
+}
+// ------ ------
+//     View
+// ------ ------
+
+// Note: It's macro so you can use it with all events.
+macro_rules! stop_and_prevent {
+    { $event:expr } => {
+        {
+            $event.stop_propagation();
+            $event.prevent_default();
+        }
+     };
+}
+trait IntoDragEvent {
+    fn into_drag_event(self) -> DragEvent;
+}
+
+impl IntoDragEvent for Event {
+    fn into_drag_event(self) -> DragEvent {
+        self.dyn_into::<web_sys::DragEvent>()
+            .expect("cannot cast given event into DragEvent")
+    }
+}
+
+fn view(model: &Model) -> Node<Msg> {
+    div![
+        generate_draggable_div(2, "I am number1", &model.drag_me_2, "green"),
+        generate_draggable_div(1, "I am number1", &model.drag_me_1, "purple"),
+        div![
+            id!("drop_zone"),
+            el_ref(&model.drop_zone),
+            "drop stuff there",
+            style![
+                St::Height => px(150),
+                St::Width => px(150),
+                St::Margin => "auto",
+                St::FontFamily => "sans-serif",
+                St::Display => "flex",
+                St::FlexDirection => "column",
+                St::TextAlign=> "center",
+                St::JustifyContent => "center",
+                St::AlignItems => "center",
+                St::Border => [&px(2), "dotted", "green"].join(" ");
+                St::BorderRadius => px(20),
+            ],
+            ev(Ev::Drop, |_| { Msg::Drop }),
+            ev(Ev::DragOver, |event| {
+                let drag_event = event.into_drag_event();
+                stop_and_prevent!(drag_event);
+                drag_event.data_transfer().unwrap().set_drop_effect("move");
+                Msg::DragOver
+            })
+        ]
+    ]
+}
+
+fn generate_draggable_div(
+    id: u8,
+    default_msg: &str,
+    div: &ElRef<HtmlDivElement>,
+    color: &str,
+) -> Node<Msg> {
+    div![
+        default_msg,
+        id!(id),
+        el_ref(div),
+        style![
+            St::Height => px(50),
+            St::Width => px(50),
+            St::FontFamily => "sans-serif",
+            St::FontSize=>"9px",
+            St::Display => "flex",
+            St::Cursor => if can_be_dragged(div) {"grab"} else {"default"},
+            St::FlexDirection => "column",
+            St::JustifyContent => "center",
+            St::AlignItems => "center",
+            St::Border => [&px(2), "solid", color].join(" ");
+            St::BorderRadius => px(20),
+        ],
+        ev(Ev::MouseOver, move |_| { Msg::MouseChange(id, true) }),
+        ev(Ev::MouseLeave, move |_| { Msg::MouseChange(id, false) }),
+        ev(Ev::DragStart, move |_| { Msg::Dragging(id) }),
+        ev(Ev::DragEnd, move |_| { Msg::DragEnded(id) })
+    ]
+}
+
+fn can_be_dragged(div: &ElRef<HtmlDivElement>) -> bool {
+    if let Some(div_drag) = div.get() {
+        div_drag.draggable()
+    } else {
+        false
+    }
```

Comment:
```suggestion
    div
        .get()
        .map(HtmlElement::draggable)
        .unwrap_or_default()
```
Just an alternative.

Other alternatives:
```rust
if let Some(div_drag) = div.get() { return div_drag.draggable() }
false
```

```rust
match div.get() {
    Some(div_drag) if div_drag.draggable() => true,
    _ => false
}
```

Reply:
Ok , I tried this one 

```rust

   div
        .get()
        .map(HtmlElement::draggable)
        .unwrap_or_default()

```

But it does not compile.

Reply:
They are just ideas to consider - you don't need to use them if you don't like them.

Ad compilation error - I didn't try to run them, maybe the compiler needs help with dereferencing or `HtmlElement` is wrong and it should be `.map(|div_drag| div_drag.draggable())` or `unwrap_or(false)` instead of `unwrap_or_default`. 

Your choice.